embedded/commander.go

```diff
@@ -0,0 +1,44 @@
+package embedded
+
+import (
+	"fmt"
+
+	"github.com/lytics/metafora/statemachine"
+)
+
+var _ statemachine.Commander = (*Commander)(nil)
+
+// Commander is an embedable statemachine.Commander implementation.
+// Task-specific command listeners are created by calling NewListener.
+type Commander struct {
+	listeners map[string]chan statemachine.Message
+}
+
+// NewCommander creates a new statemachine.Commander implementation.
+func NewCommander() *Commander {
+	return &Commander{listeners: make(map[string]chan statemachine.Message)}
+}
+
+// NewListener creates a task specific command listener linked to an embedded
+// Commander.
+func (c *Commander) NewListener(taskID string) statemachine.CommandListener {
+	// Buffer chan to make sending/recving asynchronous
+	c.listeners[taskID] = make(chan statemachine.Message, 1)
+	return &commandListener{c: c.listeners[taskID]}
+}
+
+func (c *Commander) Send(taskID string, m statemachine.Message) error {
```

Comment:
The `embedded.Commander` is local and just for use in tests. Other implementations (see `m_etcd`'s) will know how to submit commands remotely.